test/verify/run-tests

```diff
@@ -21,32 +82,124 @@ def check_valid(filename):
         return None
     return name.replace("-", "_")
 
+def build_command(filename, test, opts):
+    cmd = [filename]
+    if opts.sit:
+        cmd.append("-s")
+    if opts.trace:
+        cmd.append("-t")
+    if opts.verbosity:
+        cmd.append("-v")
+    if opts.thorough:
+        cmd.append("--thorough")
+    if not opts.fetch:
+        cmd.append("--nonet")
+    if opts.list:
+        cmd.append("-l")
+    cmd.append(test)
+    return cmd
 
 def run(opts):
-    # Now actually load the tests, any modules that start with "check-*"
-    loader = unittest.TestLoader()
-    suite = unittest.TestSuite()
+    # Build the list of tests we'll parallellize and the ones we'll run serially
+    test_loader = unittest.TestLoader()
+    parallel_tests = []
+    serial_tests = []
+    test_id = 1
+    result = 0
+    jobs = 1 if opts.list else opts.jobs
+
     for filename in glob.glob(os.path.join(os.path.dirname(__file__), "check-*")):
         name = check_valid(filename)
         if not name or not os.path.isfile(filename):
             continue
-        with open(filename, 'rb') as fp:
-            module = imp.load_module(name, fp, filename, ("", "rb", imp.PY_SOURCE))
-            suite.addTest(loader.loadTestsFromModule(module))
+        loader = importlib.machinery.SourceFileLoader(name, filename)
+        module = importlib.util.module_from_spec(importlib.util.spec_from_loader(loader.name, loader))
+        loader.exec_module(module)
+        for test_suite in test_loader.loadTestsFromModule(module):
+            for test in test_suite:
+                test_method = getattr(test.__class__, test._testMethodName)
+                test_str = "{0}.{1}".format(test.__class__.__name__, test._testMethodName)
+                if opts.tests and not any([t in test_str for t in opts.tests]):
+                    continue
+                if getattr(test_method, "_testlib__non_destructive", False):
+                    serial_tests.append(Test(test_id, build_command(filename, test_str, opts)))
```

Comment:
Please instantiate the Test once, and just conditionally append the variable to the correct list.